apps/discovery_api/lib/discovery_api/search/dataset_index.ex

```diff
@@ -0,0 +1,233 @@
+defmodule DiscoveryApi.Search.DatasetIndex do
+  @moduledoc """
+  Manages an ElasticSearch index for datasets
+  """
+  alias DiscoveryApi.Data.Model
+
+  def create_index() do
+    %{name: name, options: options} = dataset_index()
+    create_index(name, options)
+  end
+  def create_index(name, options \\ %{}) do
+    case elastic_create_index(name, options) do
+      {:ok, _} -> get_index(name)
```

Comment:
Going to move these gets out of any of the update or create stuff, as it overcomplicates this